syntax = "proto2";

package hippo.proto;

enum ErrorCode {
    // common
    ERROR_NONE = 0;
    
    // master error
    ERROR_MASTER_NOT_LEADER = 10000;
    ERROR_MASTER_SERVICE_NOT_AVAILABLE = 10001;
    ERROR_MASTER_SERIALIZE_FAIL = 10002;
    ERROR_MASTER_APP_ALREADY_EXIST = 10003;
    ERROR_MASTER_APP_NOT_EXIST = 10004;
    ERROR_MASTER_SLAVE_ALREADY_EXIST = 10005;
    ERROR_MASTER_SLAVE_NOT_EXIST = 10006;
    ERROR_MASTER_OFFLINE_SLAVE = 10007;
    ERROR_MASTER_UPDATE_SLAVE_INFO = 10008;
    ERROR_MASTER_INVALID_REQUEST = 10009;
    ERROR_MASTER_DESERIALIZE_FAIL = 10010;
    ERROR_MASTER_UPDATE_APP_FAIL = 10011;
    ERROR_MASTER_RESOURCE_CONFLICT = 10012;
    ERROR_MASTER_RESOURCE_NOT_EXIST = 10013;
    ERROR_MASTER_EXCLUDE_TEXT_CONFLICT = 10014;
    ERROR_MASTER_QUEUE_ALREADY_EXIST =  10015;
    ERROR_MASTER_QUEUE_NOT_EXIST =  10016;
    ERROR_MASTER_QUEUE_RESOURCE_DESC_ERROR = 10017;
    ERROR_MASTER_INNER_QUEUE_CANNT_DEL = 10018;
    ERROR_MASTER_RULES_WRONG_FORMAT = 10019;
    ERROR_MASTER_RULES_INVALID = 10020;
    ERROR_MASTER_OFFLINE_SLOT = 10021;
    ERROR_MASTER_MAP_RULE_NOT_EXIST = 10022;
    ERROR_MASTER_RULE_UPDATE_FAIL = 10023;
    ERROR_MASTER_RULE_RESOURCE_VALIDATE_FAIL = 10024;
    ERROR_MASTER_LABEL_CONFLICT = 10025;
    ERROR_MASTER_LABEL_NOT_EXIST = 10026;
    ERROR_MASTER_APP_SUBMIT_TO_YARN_FAIL = 10027;
    ERROR_MASTER_APP_STOP_IN_YARN_FAIL = 10028;
    ERROR_MASTER_CLUSTER_NOT_EXIST = 10029;
    ERROR_MASTER_UPDATE_SCHED_DOMAIN_ERROR = 10033;

    // resource adjust
    ERROR_RESOURCE_AJUST_NO_NEED_AJUST = 10030;
    ERROR_RESOURCE_AJUST_FAIL_NOT_ENOUGH_RESOURCE = 10031;
    ERROR_RESOURCE_AJUST_FAIL_RESOURCE_CONFLICT = 10032;

    // buddyInfo
    ERROR_BUDDY_RULE_SERIALIZE_FAIL = 10040;

    // slave error
    ERROR_SLAVE_CLEANUP_SLOT_PLAN = 20001;
    ERROR_SLAVE_SLOTID_NOT_SPECIFIED = 20002;
    ERROR_SLAVE_SLOT_NOT_EXIST = 20003;
    ERROR_SLAVE_SERIALIZE_FAIL = 20004;
    ERROR_SLAVE_SERVICE_NOT_AVAILABLE = 20005;
    ERROR_SLAVE_ACTION_FAILED = 20006;

    // data error
    ERROR_DATA_SOURCE = 30001;
    ERROR_DATA_DEST = 31001;
    ERROR_DATA_OTHER = 32001;

    ERROR_FEDERATION_CLUSTER_ALREADY_EXIST = 40001;
    ERROR_FEDERATION_CLUSTER_DESC_INVALID = 40002;
    ERROR_FEDERATION_CLUSTER_NOT_EXIST = 40003;
    ERROR_FEDERATION_CLUSTER_START_FAIL = 40004;
    ERROR_FEDERATION_CLUSTER_STOP_FAIL = 40005;

    ERROR_FEDERATION_ROUTE_POLICY_INVALID = 40010;
    ERROR_FEDERATION_ROUTE_POLICY_CONFLICT = 40011;
    ERROR_FEDERATION_ROUTE_POLICY_NOT_EXIST = 40012;
}

enum AllocateErrorCode {
    ERROR_NOCHANGE = 100;
    ERROR_SUCCESS = 200;
    ERROR_PARTIAL_SUCCESS = 201;
    ERROR_FAILURE = 202;
    ERROR_MASTER_NOTWORK = 300;
    ERROR_INVALID_REQUEST = 400;
}
